### DISCUSSION TITLE
Debugging Preview

### DISCUSSION THREAD

**Discussion by Kazagan** [QuestPDF/QuestPDF#942] [Ideas]
In version 2024.6.0 if I turn ion the debugger it will still produce the pdf with a visual of where there were issues, in 2024.6.4, it has a detailed error, but doesn't produce the pdf. Would be awesome it could do both, or have two separate settings for it. The debug information is difficult to read, while the visual is easier to do so.

**Comment by MarcinZiabek** [comment]
Thank you for sharing the feedback 😄 

I completely agree that the visualization tool is pretty handy! ❤️ 

The debug information has been introduced to provide more detailed information. Visualization often helps but does not always describe the actual root cause, rather it gives a hint where the issue could be.

I plan to surface both approaches (debug information with visualization) with the upcoming QuestPDF Companion application, a next-generation version of the Preview application. That may take a couple of months.

Would you have any suggestions on how to enhance the debug information to make it more developer-friendly or how to restructure its content to be more useful and precise?

**Reply by Kazagan** [reply]
I've only started looking at it Friday, and found it difficult initially, but most things are. I need to shift focus for a bit, when I get back to working on the PDF I will look closer at it. 
I did notice that the debug information is likely more useful once understood, and think if used in conjunction with the visual, would have the greatest benefit.

**Reply by MarcinZiabek** [reply]
I definitely agree that combining both is the most beneficial and is the target goal. In the meantime, while you are learning, I would appreciate it if you could share feedback along the way!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
